Explorer/Assets/DCL/SDKComponents/AvatarModifierArea/Systems/AvatarModifierAreaHandlerSystem.cs

```diff
@@ -0,0 +1,153 @@
+
+                triggerAreaComponent.AreaSize = pbAvatarModifierArea.Area;
+                triggerAreaComponent.IsDirty = true;
+
+                modifierAreaComponent.SetExcludedIds(pbAvatarModifierArea.ExcludeIds);
+
+                // Update effect on now excluded/non-excluded avatars
+                foreach (Transform avatarTransform in triggerAreaComponent.CurrentAvatarsInside) { CorrectAvatarHidingState(avatarTransform, modifierAreaComponent.ExcludedIds); }
+            }
+
+            foreach (Transform avatarTransform in triggerAreaComponent.ExitedThisFrame) { ToggleAvatarHiding(avatarTransform, false, modifierAreaComponent.ExcludedIds); }
+
+            foreach (Transform avatarTransform in triggerAreaComponent.EnteredThisFrame) { ToggleAvatarHiding(avatarTransform, true, modifierAreaComponent.ExcludedIds); }
+        }
+
+        [Query]
+        [All(typeof(DeleteEntityIntention), typeof(PBAvatarModifierArea))]
+        private void HandleEntityDestruction(ref CharacterTriggerAreaComponent triggerAreaComponent, ref AvatarModifierAreaComponent modifierComponent)
+        {
+            // Reset state of affected entities
+            foreach (Transform avatarTransform in triggerAreaComponent.CurrentAvatarsInside) { ToggleAvatarHiding(avatarTransform, false, modifierComponent.ExcludedIds); }
+        }
+
+        [Query]
+        [None(typeof(DeleteEntityIntention), typeof(PBAvatarModifierArea))]
+        private void HandleComponentRemoval(ref CharacterTriggerAreaComponent triggerAreaComponent, ref AvatarModifierAreaComponent modifierComponent)
+        {
+            // Reset state of affected entities
+            foreach (Transform avatarTransform in triggerAreaComponent.CurrentAvatarsInside) { ToggleAvatarHiding(avatarTransform, false, modifierComponent.ExcludedIds); }
+        }
+
+        internal void ToggleAvatarHiding(Transform avatarTransform, bool shouldHide, HashSet<string> excludedIds)
+        {
+            var found = false;
+
+            // There's no way to do a Query/InlineQuery getting both entity and TransformComponent...
+            globalWorld.Query(in AVATAR_BASE_QUERY,
+                entity =>
+                {
+                    if (found) return;
+
+                    Transform entityTransform = globalWorld.Get<AvatarBase>(entity).transform.parent;
+                    if (avatarTransform != entityTransform) return;
+
+                    found = true;
+
+                    if (globalWorld.TryGet(entity, out Profile profile) && excludedIds.Contains(profile.UserId))
+                        return;
+
+                    globalWorld.Get<AvatarShapeComponent>(entity).HiddenByModifierArea = shouldHide;
+                });
+        }
+
+        internal void CorrectAvatarHidingState(Transform avatarTransform, HashSet<string> excludedIds)
+        {
+            var found = false;
+
+            globalWorld.Query(in AVATAR_BASE_QUERY,
+                entity =>
```

Comment:
same with lambdas

Reply:
As agreed, I will create an issue to refactor that part!

Reply:
Issue created: https://github.com/decentraland/unity-explorer/issues/359